lib/spring/watcher/abstract.rb

```diff
@@ -36,14 +48,30 @@ def add(*items)
           end
         end
 
-        items = items.select(&:exist?)
+        items = items.select do |item|
+          if item.symlink?
+            item.readlink.exist?.tap do |exists|
+              debug do
+                "add: ignoring dangling symlink: #{item.inspect} -> #{item.readlink.inspect}"
+              end if !exists
```

Comment:
Not really a fan of this style of postfix conditional on a multi-line block personally (I think it's harder to spot what the conditional is doing...)